src/vs/workbench/api/node/extHost.protocol.ts

```diff
@@ -159,6 +159,9 @@ export abstract class MainThreadLanguageFeaturesShape {
 	$registerHoverProvider(handle: number, selector: vscode.DocumentSelector): TPromise<any> { throw ni(); }
 	$registerDocumentHighlightProvider(handle: number, selector: vscode.DocumentSelector): TPromise<any> { throw ni(); }
 	$registerReferenceSupport(handle: number, selector: vscode.DocumentSelector): TPromise<any> { throw ni(); }
+	// TODO(nick): Should this support sending more than one location at a time?
```

Comment:
The only reason was to avoid confusion that this progress doesn't pass back all results every time (just intermediate progress). That can be solved with documentation though so I agree that Location[] would be more flexible.